### PR TITLE
adding monadError instance for Future on 7.1.x

### PR DESCRIPTION
**Pull request by vmarquez**

### CODE REVIEW

File: core/src/main/scala/scalaz/std/Future.scala [hunk 1]
```diff
@@ -5,7 +5,8 @@ import scala.concurrent.{Await, CanAwait, ExecutionContext, Future}
 import scala.concurrent.duration.Duration
 
 trait FutureInstances1 {
-  implicit def futureInstance(implicit ec: ExecutionContext): Monad[Future] with Cobind[Future] =
+
+  implicit def futureInstance(implicit ec: ExecutionContext): Cobind[Future] with MonadError[({type λ[α,β] = Future[β]})#λ, Throwable] with Catchable[Future] =
```
Comment (S11001001) [inline core/src/main/scala/scalaz/std/Future.scala]: I *think* you can make this bin-compatible by putting `Monad[Future]` first again. Yeah it's weird.

Comment (vmarquez) [inline core/src/main/scala/scalaz/std/Future.scala]: @S11001001 I tried this and no luck.  I can keep playing around, but any other suggestions are appreciated. :)

Comment (S11001001) [inline core/src/main/scala/scalaz/std/Future.scala]: @vmarquez Eh. Well, how about, give your signature a new name, and take off the `implicit` keyword from `futureInstance` and deprecate it.  It's a simpler case than #582; no new priority layers are needed, your instance should go where the current one does.  That will still yield a mima warning, but might be except-able.

Comment (vmarquez) [inline core/src/main/scala/scalaz/std/Future.scala]: I'm still a bit stuck, perhaps @xuwei-k or you can help me understand further. When I tried to *add* a method to FutureInstances I got this error

```
[error]  * method newFutureInstance(scala.concurrent.ExecutionContext)scalaz.Monad in trait scalaz.std.FutureInstances is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("scalaz.std.FutureInstances.newFutureInstance")
java.lang.RuntimeException: scalaz-core: Binary compatibility check failed!
```

Comment (xuwei-k) [inline core/src/main/scala/scalaz/std/Future.scala]: There are two kinds; **scalaz datatype** and **std datatypes**.

**scalaz datatypes**: We can define type class instances in companion object or its super **classes** (eg. `sealed abstract class KleisliInstances9`). We can add new methods in companion object or super classes with keep binary compatibility.

**std datatypes**: We can't define type class instances in companion object. We **should** define in **trait** (not class) due to scalaz has some import style
- _a la carte_ import `import scalaz.std.scalaFuture._`
- `import scalaz.std.AllInstances._`
- `import scalaz.Scalaz._`

We **can't** add new methods in trait with keep binary compatibility (scala 2.11 or earlier)


In other words, adding new **std datatype** instances is harder than **Scalaz datatype** in scalaz7 😔 


Here is a unfortunate way add new typeclass instance.
https://github.com/xuwei-k/scalaz/commit/9c246bd5cfeed81bb9bc6d0f1e586f321a946388

Comment (S11001001) [inline core/src/main/scala/scalaz/std/Future.scala]: @vmarquez This error should be except-able, because we only support the three `import` lines @xuwei-k mentions, and all these objects will be recompiled to include the method you've added.

**Comment by vmarquez** [comment]
Thank you @xuwei-k and @S11001001 for helping me.  I will hopefully be able to help back port more stuff now...
